public/table.js

```diff
@@ -0,0 +1,106 @@
+                        TableController.stringCompare(a[column], b[column], order) :
+                        TableController.sortNumber(a[column], b[column], order)
+                );
+
+            // Table Cells
+            let td = tr.selectAll("td")
+                .data(d => TableController.jsonToArray(d))
+                .enter().append("td")
+                .on("click", (d, i, n) => transform(d[0]));
+
+            // Filter Dataset column from columns & append text to td
+            td.filter((d, i, n) => d[0] !== "Dataset" && i !== 0)
+                .text(d => ["Done", "WS %", "NC %"].includes(d[0]) ? `${d[1]}%` : d[1])
+
+            // Select Dataset Column, create a link & append text to td
+            td.filter((d, i, n) => d[0] === "Dataset" && i === 0)
+                .append("a")
+                .attr("href", d => `https://web-coda.firebaseapp.com/?dataset=${d[1]}`)
+                .attr("target", "_blank")
+                .text(d => d[1])
+
+            // Filter table to remain with "Done" column
+            td.filter((d, i, n) => d[0] === "Done" && i === 3)
+            .each((d, i, n) => {
+                // Select Table Row
+                let parentNode = d3.select(n[i].parentNode)
+                // Select Table Data and access data bound to the node
+                let tableData = d3.select(n[i]).data()[0][1]
+                if (parseFloat(tableData) === 0) {
+                    parentNode.attr('class', "coding-notstarted");
+                } else if (parseFloat(tableData) > 0 && parseFloat(tableData) <= 25) {
+                    parentNode.attr('class', "coding-below25");
+                } else if (parseFloat(tableData) > 25 && parseFloat(tableData) <= 50) {
+                    parentNode.attr('class', "coding-above25");
+                } else if (parseFloat(tableData) > 50 && parseFloat(tableData) <= 75) {
+                    parentNode.attr('class', "coding-above50");   
+                } else if (parseFloat(tableData) > 75 && parseFloat(tableData) < 100) {
+                    parentNode.attr('class', "coding-above75");  
+                } else {
+                    parentNode.attr('class', "coding-complete");
+                }
+            })
+        };
+    };
+
+    static stringCompare(a, b, order) {
+        if (order === "descending") 
+            return a.localeCompare(b, 'en', { sensitivity: 'base' });
+        return b.localeCompare(a, 'en', { sensitivity: 'base' });
+    };
+
+    static sortNumber(a,b, order) {
+        if (order === "descending") 
+            return a-b || isNaN(a)-isNaN(b);
+        return b-a || isNaN(b)-isNaN(a);
+    } 
+
+    static jsonKeyValueToArray(k, v) {return [k, v];}
+
+    static jsonToArray(json) {
+        var arr = [];
+        for (const key in json) {
+            if (json.hasOwnProperty(key)) {
+                arr.push(TableController.jsonKeyValueToArray(key, json[key]));
+            }
+        }
+        return arr;
+    };
+}
```

Comment:
nit: add a new line at the end you can also setup vs-code to always do this for you  https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files

Reply:
Thanks, I've enable it